modules/4337/contracts/Safe4337Module.sol

```diff
@@ -212,9 +212,17 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     }
 
     /**
-     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not
+     * @notice Checks if the signatures length is correct and does not contain additional bytes. The function does not
      * check the integrity of the signature encoding, as this is expected to be checked by the {Safe} implementation
      * of {checkSignatures}.
+     * @dev A malicious bundler can pad additional bytes to the `signatures` data, causing the account to pay more gas
+     * than needed for user operation validation. Safe account has two types of signatures: EOA and Smart Contract
+     * signatures. While the EOA signature is fixed in size, the Smart Contract signature can be of arbitrary length.
+     * Safe encodes the Smart Contract signature length in the signature data. Since, the `signature` field in UserOp
+     * is not part of the UserOp hash a malicious bundler can manipulate the field storing the signature length and pad
```

Comment:
Doubt: I thought only the signature offset and additional bytes added to the dynamic part could be manipulated. Won't a change in signature length change the actual signature taken by the Safe Contracts for verification, thus resulting in a definite failure of signature verification?

Reply:
It depends on the account implementation whether the signature verification will fail or pass. In case of Safe, it will pass.

The test case here will fail because verification passed though there were padded bytes in signature. 
https://github.com/safe-global/safe-modules/commit/c825117819b5ddc3d61a503745b44b441f061282